### DISCUSSION TITLE
Multirun:Sweep Get Current Job Config globally from inside the user code

### DISCUSSION THREAD

**Discussion by yogi81** [facebookresearch/hydra#2403] [Q&A]
Hello,

when my job is running from a `BasicSweeper` how can I access the current active config from within my code being currently executed?

It is possible to access hydra part of the config with `hydra.core.hydra_config.HydraConfig.get()` using the singleton pattern.
But the returned dict contains only hydra: settings and lacks user defined variables/configurations

Is there any singleton/globals based method to retrieve user define configuration in a same way?

Thanks a lot in advance!

**Comment by Jasha10** [comment]
Hi,

Doesn't your `@hydra.main`-decorated function take your user-defined `cfg` as a parameter?
> how can I access the current active config

```python
...

@hydra.main(...)
def my_app(cfg):  #  <- cfg is the current active config
   ...
```

**Reply by awav** [reply]
I guess, @yogi81 means that `cfg` doesn't have `hydra` configuration.

**Comment by omry** [comment]
There is no facility to access the user config via a singleton and there wouldn't be one.
You need to either pass it explicitly (recommended) or implement your own singleton solution.
